rollup/plugins/extractPolyfillsPlugin.js

```diff
@@ -0,0 +1,208 @@
+    if (declarationId.type === 'Identifier') {
+      return declarationId.name;
+    }
+    // Declarations of the form
+    //   `const { dogs } = { dogs: function() { return "are great"; } }`
+    // or
+    //   `const { dogs } = { dogs: () => "are great" }`
+    else if (declarationId.type === 'ObjectPattern') {
+      return declarationId.properties[0].key.name;
+    }
+    // Any other format
+    else {
+      return 'unknown variable';
+    }
+  }
+
+  // Regular old functions, of the form
+  //   `function dogs() { return "are great"; }`
+  else if (node.type === 'FunctionDeclaration') {
+    return node.id.name;
+  }
+
+  // If we get here, this isn't a node we're interested in, so just return a string we know will never match any of the
+  // polyfill names
+  else {
+    return 'nope';
+  }
+}
+
+/**
+ * Find all nodes whose identifiers match a known polyfill name.
+ *
+ * Note: In theory, this could yield false positives, if any of the magic names were assigned to something other than a
+ * polyfill function, but the chances of that are slim. Also, it only searches the module global scope, but that's
+ * always where the polyfills appear, so no reason to traverse the whole tree.
+ */
+function findPolyfillNodes(ast) {
+  const isPolyfillNode = node => {
+    const nodeName = getNodeName(node);
+    if (POLYFILL_NAMES.has(nodeName)) {
+      // mark this node for later deletion, since we're going to replace it with an import statement
+      node.shouldDelete = true;
+      // store the name in a consistent spot, regardless of node type
+      node.name = nodeName;
+
+      return true;
+    }
+
+    return false;
+  };
+
+  return ast.program.body.filter(isPolyfillNode);
+}
+
+/**
+ * Create a node representing an `import` or `require` statement of the form
+ *
+ *     import { < polyfills > } from '...'
+ * or
+ *     var { < polyfills > } = require('...')
+ *
+ * @param polyfillNodes The nodes from the current version of the code, defining the polyfill functions
+ * @param currentSourceFile The path, relative to `src/`, of the file currently being transpiled
+ * @param moduleFormat Either 'cjs' or 'esm'
+ * @returns A single node which can be subbed in for the polyfill definition nodes
+ */
+function createImportOrRequireNode(polyfillNodes, currentSourceFile, moduleFormat) {
+  const {
+    callExpression,
+    identifier,
+    importDeclaration,
+    importSpecifier,
+    literal,
+    objectPattern,
+    property,
+    variableDeclaration,
+    variableDeclarator,
+  } = recast.types.builders;
+
+  // Since our polyfills live in `@sentry/utils`, if we're importing or requiring them there the path will have to be
+  // relative
+  const isUtilsPackage = process.cwd().endsWith('packages/utils');
+  const importSource = literal(
+    isUtilsPackage
+      ? path.relative(path.dirname(currentSourceFile), `../jsPolyfills/${moduleFormat}`)
+      : `@sentry/utils/jsPolyfills/${moduleFormat}`,
```

Comment:
should we make another package for the polyfills?

Reply:
> In the long term, they might well live in a `@sentry/dev-utils`/`@sentry/internal-utils`/`@sentry/whatever package`, but to get things running, for now the rollup directory seemed like a good place

In the long run I agree that'd be a better place, but I'd say it's okay to leave this as an iterative improvement. What do you think?

Reply:
I'm +1 on an extra package but it's probably fine if we do it in a follow-up task (whenever).

Reply:
I don't think they warrant a package all on their own, but I did experiment with moving them to a `dev-utils` package. Once I got them there, though, I didn't love that it was then going to become a runtime dependency of `@sentry/utils`, when it really should only be a dev dependency. And as part of that experimentation, I did end up translating them to TS, which was the final remaining blocker to them living directly in `@sentry/utils`, so that is where they ended up.

(The other two original blockers for me were:
- Feeling weird about putting code that wasn't ours in with code that was. This was solved for me by 1) the fact that we now go overboard in terms of attribution, license-copying, etc, and 2) the fact that in fact they aren't exact copies of the originals anymore, if for no other reason than the translation to TS.
- Not wanting them to be part of the public API of `@sentry/utils`. Originally that meant that I was copying them over to a folder outside of `build`, but I realized that their built versions actually can live inside `build` and still be "hidden" (in that they won't show up in intellisense completions, for example) as long as they're not exported from `index.ts`. That then let me put the folder containing the pre-build versions straight into `src/`, which made everything about building and testing them a lot easier.)